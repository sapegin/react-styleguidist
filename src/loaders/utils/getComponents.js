const processComponent = require('./processComponent').default;

/**
 * Process each component in a list.
 *
 * @param {Array} components File names of components.
 * @param {object} config
 * @returns {object|null}
 */
module.exports = function getComponents(components, config) {
	return components.map(filepath => processComponent(filepath, config));
};
