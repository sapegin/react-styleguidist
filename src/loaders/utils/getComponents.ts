import processComponent from './processComponent';
import * as Rsg from '../../typings';

/**
 * Process each component in a list.
 *
 * @param components File names of components.
 * @param config
 */
export default function getComponents(
	components: string[],
	config: Rsg.SanitizedStyleguidistConfig
) {
	return components.map(filepath => processComponent(filepath, config));
}
