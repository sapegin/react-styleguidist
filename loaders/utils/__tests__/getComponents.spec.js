import identity from 'lodash/identity';
import getComponents from '../getComponents';

it('getComponents() should return an object for components', () => {
	const result = getComponents(['Foo.js', 'Bar.js'], {
		configDir: require.resolve('_test'),
		getExampleFilename: identity,
		getComponentPathLine: identity,
	});

	expect(result).toMatchSnapshot();
});
