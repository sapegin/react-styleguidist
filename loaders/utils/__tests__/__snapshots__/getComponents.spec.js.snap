exports[`test getComponents() should return an object for components 1`] = `
Array [
  Object {
    "examples": null,
    "filepath": "../Foo.js",
    "module": RequireStatement {
      "filepath": "Foo.js",
    },
    "nameFallback": "Foo",
    "pathLine": "../Foo.js",
    "props": RequireStatement {
      "filepath": "!!<rootDir>/loaders/props-loader.js!Foo.js",
    },
  },
  Object {
    "examples": null,
    "filepath": "../Bar.js",
    "module": RequireStatement {
      "filepath": "Bar.js",
    },
    "nameFallback": "Bar",
    "pathLine": "../Bar.js",
    "props": RequireStatement {
      "filepath": "!!<rootDir>/loaders/props-loader.js!Bar.js",
    },
  },
]
`;
